elyra/metadata/schemasproviders.py

```diff
@@ -29,6 +29,20 @@
     # We may not have kfp-tekton available and that's okay!
     TektonClient = None
 
+try:
+    from kfp import compiler  # noqa
+except ImportError:
+    # KFP python package is not installed
+    compiler = None
+
+try:
+    # Check to see if Airflow package is installed, since we do not have any dependencies
+    # on any Airflow package, we use GitHub as our canary
+    from github import Github  # noqa
```

Comment:
I don't think this will be sufficient.  Won't we still need the ability for the user to physically indicate their intentions?  Like `pip install elyra[kfp]` or `pip install elyra[airflow]`.

Do we need a dummy airflow package that we could use (and install via the previous example) to make the user's intention known to server code?  This package, while on PyPi and Conda-forge, would likely never have to change, so I'm not sure there would be much of a burden other than its initial creation.

Then `elyra[kfp]` would include the KFP sdk, `elyra[kfp-tekon]` would include `elyra[kfp] + kfp-tekton` and `elyra[airflow]` would include this dummy airflow placeholder as its dependency.  It could even have a dependency on GitHub, if we needed that (and `elyra[gitlab]` could include `elyra[airflow] + gitlab` if we wanted that package tied to airflow.

Reply:
I agree. There may be other reasons a package (like github) might be installed. For example, once the community provides a component catalog connector that relies on the package.